hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson

```diff
@@ -0,0 +1,291 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  name: rv_dm
+  testpoints: [
+    // RV_DM (JTAG) tests:
+    {
+      name: chip_jtag_csr_rw
+      desc: '''
+            Verify accessibility of all the CSRs in the chip over JTAG.
+
+            - Shuffle the list of CSRs first to remove the effect of ordering.
+            - Write all CSRs via JTAG interface with a random value.
+            - Shuffle the list of CSRs yet again.
+            - Read all CSRs back and check their values for correctness while adhering to the CSR's
+              access policies.
+            - Accesses to CSRs external to `rv_dm` go through RV_DM SBA interface into the `xbar`.
+            '''
+      stage: V2
+      si_stage: SV2
+      tests: ["chip_jtag_csr_rw"]
+      lc_states: ["TEST_UNLOCKED", "DEV", "RMA"]
+      host_support: "true"
+      features: [
+        "RV_DM.JTAG.FSM",
+        "RV_DM.JTAG.DTM"
+      ]
+    }
+    {
+      name: chip_jtag_mem_access
+      desc: '''
+            Verify accessibility of all the memories in the chip over JTAG.
+
+            This test will target the following memories in the chip:
+              sram_main, sram_ret, otbn i|dmem, ROM
+
+            - Shuffle the list of memories first to remove the effect of ordering.
+            - Write a location in a randomly chosen set of addresses within each memory via JTAG
+              interface with random values.
+            - For read-only memories:
+               - In simulation preload the memory with random data via backdoor.
+               - For silicon choose some known values and addresses in these memories and shuffle
+                 these to choose one from each for the test.
+            - Shuffle the list of memories again.
+            - Read the previously written addresses in the memories back again and check the read
+              value for correctness. Pick some random addresses to verify in case of read-only
+              memories.
+            '''
+      stage: V2
+      si_stage: SV2
+      tests: ["chip_jtag_mem_access"]
+      lc_states: ["TEST_UNLOCKED", "DEV", "RMA"]
+      host_support: "true"
+      features: [
+        "RV_DM.JTAG.FSM",
+        "RV_DM.JTAG.DTM",
+        "RV_DM.DBG.SBA"
+      ]
+    }
+    {
+      name: chip_rv_dm_perform_debug
+      desc: '''
+            - X-ref'ed with rom_e2e_jtag_inject from rom testplan.
+            - X-ref'ed with chip_sw_flash_lc_iso_part_sw_wr_en.
+            - X-ref'ed with manuf_cp_device_info_flash_wr from manufacturing testplan.
+            - Using the sram injection mechanism from rom_e2e_jtag_inject, load a SRAM program that
+              writes to isolated flash partition while the device is in TEST_UNLOCKED state.
+            - After writing, verify that the test program cannot read back the written value.
```

Comment:
It is not clear to me what this means. The SRAM program writes something to the flash. Then the `test program` (is this another program that is bootstraped?) needs to try and read it?